odl/tomo/geometry/conebeam.py

```diff
@@ -116,49 +686,167 @@ def __init__(self, apart, dpart, src_radius, det_radius, pitch,
+        """Create an instance of `HelicalConeFlatGeometry` using a matrix.
+
+        This alternative constructor uses a matrix to rotate and
+        translate the default configuration. It is most useful when
+        the transformation to be applied is already given as a matrix.
+
+        Parameters
+        ----------
+        apart : 1-dim. `RectPartition`
+            Partition of the parameter interval.
+        dpart : 2-dim. `RectPartition`
+            Partition of the detector parameter set.
+        src_radius : nonnegative float
+            Radius of the source circle.
+        det_radius : nonnegative float
+            Radius of the detector circle.
+        pitch : float
+            Constant vertical distance that a point on the helix
+            traverses when increasing the angle parameter by ``2 * pi``.
+        init_matrix : `array_like`, shape ``(3, 3)`` or ``(3, 4)``, optional
+            Transformation matrix whose left ``(3, 3)`` block is multiplied
+            with the default ``det_pos_init`` and ``det_axes_init`` to
+            determine the new vectors. If present, the fourth column acts
+            as a translation after the initial transformation.
+            The resulting ``det_axes_init`` will be normalized.
+        pitch_offset : float, optional
+            Offset along the ``axis`` at ``angle=0``. Default: 0.
+
+        Returns
+        -------
+        geometry : `HelicalConeFlatGeometry`
+            The resulting geometry.
+
+        Examples
+        --------
+        Map unit vectors ``e_y -> e_z`` and ``e_z -> -e_y``, keeping the
+        right-handedness:
+
+        >>> apart = odl.uniform_partition(0, 2 * np.pi, 10)
+        >>> dpart = odl.uniform_partition([-1, -1], [1, 1], (20, 20))
+        >>> matrix = np.array([[1, 0, 0],
+        ...                    [0, 0, -1],
+        ...                    [0, 1, 0]])
+        >>> geom = HelicalConeFlatGeometry.frommatrix(
+        ...     apart, dpart, src_radius=5, det_radius=10, pitch=2,
+        ...     init_matrix=matrix)
+        >>> geom.axis
+        array([ 0., -1.,  0.])
+        >>> geom.src_to_det_init
+        array([ 0.,  0.,  1.])
+        >>> geom.det_axes_init
+        (array([ 1.,  0.,  0.]), array([ 0., -1.,  0.]))
+
+        Adding a translation with a fourth matrix column:
+
+        >>> matrix = np.array([[0, 0, -1, 0],
+        ...                    [0, 1, 0, 1],
+        ...                    [1, 0, 0, 1]])
+        >>> geom = HelicalConeFlatGeometry.frommatrix(
+        ...     apart, dpart, src_radius=5, det_radius=10, pitch=2,
+        ...     init_matrix=matrix)
+        >>> geom.translation
+        array([ 0.,  1.,  1.])
+        >>> geom.det_refpoint(0)  # (0, 10, 0) + (0, 1, 1)
+        array([  0.,  11.,   1.])
+        """
+        for key in ('axis', 'src_to_det_init', 'det_axes_init', 'translation'):
+            if key in kwargs:
+                raise TypeError('got unknown keyword argument {!r}'
```

Comment:
This follows common Python error message style for invalid arguments.
But anyway, it's probably too much checking anyway, since the call into `__init__` explicitly sets these parameters, so specifying them again in `kwargs` will cause an exception.